WIP: Fetches and use specific version of Google Test from source

jilpi opened the pull request:
Will fetch all sources from https://github.com/google/googletest
Tested with Google Test 1.8.0 and 1.10.0
Currently selected version is 1.10.0

Google Test version can be changed here:
https://github.com/NimbleSM/NimbleSM/blob/feature/google-test/unit_tests/CMakeLists.txt#L10

Closes issue #88

Commit:
Merge remote-tracking branch 'origin/master' into feature/centralize-tests

Commit:
Added "np1/2" suffix to test names

Commit:
Fetch Google Test from Github and build from source

Can use a specific version of Google Test. All current unit tests can pass.

Commit:
Update to Google Test v1.10.0

Removes warning, all tests can pass

Commit:
Added unit tests to CI

Commit:
Unit tests can now run without Kokkos

uhetmaniuk reviewed (commented):
Looks good.
Will let @nmm0 review before merging

Commit:
Added support to cmake 3.12 (<3.14)

jilpi commented:
Hi @nmm0 
You're right. Added support for cmake 3.12, which is the current minimum version for NimbleSM.
Can you give it another look?

nmm0 commented on unit_tests/CMakeLists.txt:
```diff
@@ -18,7 +18,16 @@ FetchContent_Declare(
+if (${CMAKE_VERSION} VERSION_LESS "3.14")
+  message("Cmake < 3.14 - consider upgrading (FetchContent_MakeAvailable())")
+  FetchContent_GetProperties(googletest)
+  if(NOT googletest_POPULATED)
```
I'm not 100% sure this will work since CMake is a bit weird with versions. because we require CMake 3.12, newer versions may downgrade themselves to 3.12 when executing the script. It's simpler just to use the add_subdirectory approach since it works on newer versions of CMake and it's basically what MakeAvailable does just a bit more verbose

Commit:
Code focused on CMake 3.12 features

Removed all code referencing features implemented in cmake 3.14 to ensure perfect compatibility

jilpi commented:
@nmm0 Commented all code related to 3.14 (available for future reference), and kept only code working with 3.12+

nmm0 commented:
Closing this as it seems that these changes were already merged in
